packages/application-shell/src/components/application-shell-authenticated/application-shell-authenticated.tsx

```diff
@@ -201,263 +215,320 @@ export const ApplicationShellAuthenticated = (
                     flags={props.featureFlags}
                     defaultFlags={props.defaultFeatureFlags}
                   >
-                    <>
-                      <ThemeSwitcher />
-                      <VersionTracker />
-                      {/* NOTE: the requests in flight loader will render a loading
-                      spinner into the AppBar. */}
-                      <RequestsInFlightLoader />
-                      <SentryUserTracker user={user ?? undefined} />
-                      <div
-                        css={css`
-                          height: 100vh;
-                          display: grid;
-                          grid-template-rows: auto ${DIMENSIONS.header} 1fr;
-                          grid-template-columns: auto 1fr;
-                        `}
-                      >
-                        <div
-                          ref={notificationsGlobalRef}
-                          role="region"
-                          aria-live="polite"
-                          css={css`
-                            grid-row: 1;
-                            grid-column: 1/3;
-                          `}
-                        >
-                          <div id="above-top-navigation" />
-                          <NotificationsList domain={DOMAINS.GLOBAL} />
-                        </div>
-
-                        <Route>
-                          {() => {
-                            if (!projectKeyFromUrl) return <QuickAccess />;
-                            return (
-                              <FetchProject projectKey={projectKeyFromUrl}>
-                                {({ isLoading: isProjectLoading, project }) => {
-                                  if (isProjectLoading || !project) return null;
-
-                                  // when used outside of a project context,
-                                  // or when the project is expired or supsended
-                                  const shouldUseProjectContext = !(
-                                    (project.suspension &&
-                                      project.suspension.isActive) ||
-                                    (project.expiry && project.expiry.isActive)
-                                  );
+                    <NewMainNavigationFlagWrapper>
+                      {(isNewNavigationEnabled: boolean) => {
```

Comment:
This wrapper was necessary to pass the feature flag prop to children, since we need to evaluate the flag after the FlopFlip provider is rendered.